pkg/config/piped.go

```diff
@@ -590,6 +593,23 @@ type PipedImageWatcher struct {
 	Repos []PipedImageWatcherRepoTarget `json:"repos"`
 }
 
+// Validate checks if the duplicated repository setting exists,
+// and if both includes and excludes are given in each repo.
+func (i *PipedImageWatcher) Validate() error {
+	repos := make(map[string]struct{})
+	for _, repo := range i.Repos {
+		if _, ok := repos[repo.RepoID]; ok {
+			return fmt.Errorf("duplicated repo id (%s) found in the imageWatcher directive", repo.RepoID)
+		}
+		repos[repo.RepoID] = struct{}{}
+
+		if len(repo.Includes) != 0 && len(repo.Excludes) != 0 {
+			return fmt.Errorf("both includes and excludes are given in %s", repo.RepoID)
```

Comment:
Do we have any reasons for not allowing this?

Reply:
Our recognition might be different. I'm thinking Includes is like a whitelist, and Excludes is like a blacklist.
- When using Excludes, include all files that are not specified
- When using Includes, include all specified files

If both are given, we can't decide which to prioritize.

Reply:
I got it. These are lists of filenames right, not regexes.
IMO, we can ignore this error by taking the `excludes` with a higher priority.

Reply:
I'm unfamiliar with the standard, but is it general to prioritize the Excludes? I'm worried about letting users be confused.

Reply:
I think there are no rules for that.
(or just a ref: https://gist.github.com/ento/cf5593d4fac3f72d3765b3fe2ed2a204)

But I think in our case, by missing/wrong-configuration, `no deployment was triggered` is better/safer than `a deployment was triggered`.

Reply:
I agree with that, okay, let's make Excludes top priority and will tell about the priority to users well.